homeassistant/components/esphome/__init__.py

```diff
@@ -845,6 +845,7 @@ async def async_added_to_hass(self) -> None:
                 self._on_static_info_update,
             )
         )
+        self._update_state_from_entry_data()
```

Comment:
If state comes in before entity info we didn't set it.  This doesn't actually happen in practice though but the api doesn't disallow it